src/box/lua/config/instance_config.lua

```diff
@@ -1118,6 +1118,67 @@ return schema.new('instance_config', schema.record({
             end
         end,
     }),
+    security = schema.record({
+        auth_type = schema.enum({
+            'chap-sha1',
+            'pap-sha256',
+        }, {
+            box_cfg = 'auth_type',
+            default = 'chap-sha1',
+            validate = function(auth_type, w)
+                if auth_type ~= 'chap-sha1' and
+                        tarantool.package ~= 'Tarantool Enterprise' then
+                    w.error('chap-sha1 is the only auth_type available in' ..
+                            ' Tarantool CE')
```

Comment:
I think it is better to write something like `The PAP-SHA256 authentication method is only available in Tarantool Enterprise Edition`. This way it will be similar to our usual `This configuration parameter is available only in Tarantool Enterprise Edition`.

Reply:
I think it's better to describe what a user would get in EE than to say what we have in CE.

Reply:
1. CE/EE abbreviations shouldn't be used in error messages, because it is not obvious for a user who isn't get in touch with the project hard enough to know our internal teminology. Let's use explicit 'Tarantool Community Edition' and 'Tarantool Enterprise Edition'.
2. It would be good to mention the problematic value in the error message.
3. I would prefer to show configuration values as is in quotes (use `%q`) and use human-readable option description to point the option ('authentication method' instead of 'auth_type').
4. The new authentication method may be added into CE, EE, both or moved from EE to CE. We anyway should reflect it in the configuration schema.

To sum up, I mostly support Mergen's proposal.

Reply:
@Totktonada, I mostly agree with you, see comments below:
1) Fixed.
2) Done
3) Agreed.
4) The validation function shouldn't really have such info other than the list of options that are valid. Now there is only one right option, so there is no need for complexity. I don't mind it being added in the future with other `auth_type`-s.

```lua
              validate = function(auth_type, w)
                  if auth_type ~= 'chap-sha1' and
                          tarantool.package ~= 'Tarantool Enterprise' then
                          w.error("'chap-sha1' is the only auth_type available in" .. 
                                  " Tarantool Community Edition (%q requested)",
                                  auth_type)                                                                  
                  end                                                                                                                                                              
              end                                                                                                                                                                                                                                                                                                                                
```

Reply:
It is good enough for my taste, but I would use 'authentication method' instead of 'auth_type' (see point 3 above).